### ISSUE TITLE
Consider adding a Lifecycle param to the PaymentClient to auto-close connections on destroy

### ISSUE THREAD

**Issue by andreas-borglin** [Aevi-UK/pos-flow-sdk#85]
To avoid
```Activity com.aevi.android.rxmessenger.sample.ClientActivity has leaked ServiceConnection com.aevi.android.rxmessenger.client.MessengerConnection@f281989 that was originally bound here```
when an activity is destroyed without explicitly closing the service connection

**Comment by brettwold** [comment]
New RxMessenger library release will allow FPS to detect clients disappearing automatically

**Comment by andreas-borglin** [comment]
Yes but that will only fix the issue from the FPS end - this would ensure we don't get all those leaking warnings from the client when an activity is destroyed with a hanging service connection

**Comment by andreas-borglin** [comment]
Let's test and validate whether the connection is closed or not when the activity is destroyed without disposing the disposable.

**Comment by andreas-borglin** [comment]
We need to update samples and instruct via docs that clients always dispose in onDestroy
